messaginginapp/src/main/java/io/customer/messaginginapp/ModuleMessagingInApp.kt

```diff
@@ -85,38 +79,47 @@ internal constructor(
     private fun setupGistCallbacks() {
         gistProvider.subscribeToEvents(onMessageShown = { deliveryID ->
             logger.debug("in-app message shown $deliveryID")
-            trackRepository.trackInAppMetric(
-                deliveryID = deliveryID,
-                event = MetricEvent.opened
+            eventBus.publish(
+                Event.TrackInAppMetricEvent(
+                    deliveryID = deliveryID,
+                    event = MetricEvent.opened.name
+                )
             )
         }, onAction = { deliveryID: String, _: String, action: String, name: String ->
             logger.debug("in-app message clicked $deliveryID")
-            trackRepository.trackInAppMetric(
-                deliveryID = deliveryID,
-                event = MetricEvent.clicked,
-                metadata = mapOf("action_name" to name, "action_value" to action)
+            eventBus.publish(
+                Event.TrackInAppMetricEvent(
+                    deliveryID = deliveryID,
+                    event = MetricEvent.clicked.name,
+                    params = mapOf("action_name" to name, "action_value" to action)
+                )
             )
         }, onError = { errorMessage ->
             logger.error("in-app message error occurred $errorMessage")
         })
     }
 
     private fun setupHooks() {
-        hooksManager.add(
-            module = HookModule.MessagingInApp,
-            subscriber = ModuleInAppHookProvider()
-        )
+        eventBus.subscribe<Event.ScreenViewedEvent> {
+            gistProvider.setCurrentRoute(it.name)
+        }
+
+        eventBus.subscribe<Event.ProfileIdentifiedEvent> {
+            gistProvider.setUserToken(it.identifier)
+        }
+
+        eventBus.subscribe<Event.ResetEvent> {
+            gistProvider.clearUserToken()
+        }
     }
 
+    // TODO: Remove config and replace it with moduleConfig
     private fun initializeGist(config: CustomerIOConfig) {
         gistProvider.initProvider(
-            application = diGraph.context.applicationContext as Application,
+            // TODO: This should not be nullable
+            application = diGraph.androidSDKComponent?.applicationContext as Application,
```

Comment:
Similar to [our earlier discussion](https://github.com/customerio/customerio-android/pull/359#discussion_r1638943077). Outside of this PR though, but I think we need to improve how we use `AndroidSDKComponent` in our modules as they all have public constructors like you pointed out.

For this line, I think we need to use safe cast?
```suggestion
            application = diGraph.androidSDKComponent?.applicationContext as? Application,
```

Or may be use `let` or early exit here as `GistProvider` may not allow nullable Application instance.

Reply:
`AndroidSDKComponent` needs to be non-nullable otherwise we will keep on hitting this but as you pointed out, future update